src/main/java/com/salesforce/dynamodbv2/mt/mappers/index/DynamoSecondaryIndexMapperTrackingAssigned.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.dynamodbv2.mt.mappers.index;
+
+import com.salesforce.dynamodbv2.mt.mappers.MappingException;
+import com.salesforce.dynamodbv2.mt.mappers.metadata.DynamoTableDescription;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A secondary index mapper that keeps track of what the previous virtual-to-physical index assignments were, so that
+ * assigned physical indexes are not reused.
+ */
```

Comment:
The purpose of this is to allow mapping multiple GSIs of a virtual table to compatible GSIs of a physical table, when the physical GSIs have the same PK types, since with hash partitioning all PKs have a binary HK and a binary RK, and the default test setup has virtual tables with multiple GSIs (each with different PK types).

Reply:
We assume the order in which virtual secondary indexes are defined is always the same. If not then this is not deterministic. Ideally we'd persist the virtual-to-physical index mapping in the table description repo, and I started looking into that, but didn't do it... I thin we'll need that in order to support updating existing tables.